internal/pkg/server/server.go

```diff
@@ -82,6 +120,201 @@ func (s *Server) postTar(c echo.Context) error {
+	return nil
+}
+
+func (s *Server) postHelmChart(c echo.Context) error {
+	chrtRef := c.FormValue("chart")
+	chrtVer := c.FormValue("version")
+	file, err := c.FormFile("values.yaml")
+	if err != nil {
+		return err
+	}
+	src, err := file.Open()
+	if err != nil {
+		return err
+	}
+	defer src.Close()
+	buf, err := ioutil.ReadAll(src)
+	if err != nil {
+		return err
+	}
+	values := map[string]interface{}{}
+	if err := yaml.Unmarshal(buf, &values); err != nil {
+		return err
+	}
+	var authn auth.Authenticator
+	_, ok := c.QueryParams()["auth"] // Currently support only docker config.json
+	if ok {
+		var err error
+		authn, err = authFromFormFile(c, "config.json")
+		if err != nil {
+			return err
+		}
+	} else {
+		authn = auth.EmptyAuthenticator
+	}
+	_, verify := c.QueryParams()["verify"]
+	manifest, err := renderHelmChart(chrtRef, chrtVer, values, verify)
+	if err != nil {
+		return err
+	}
+	images, err := findImagesInManifest(manifest)
+	if err != nil {
+		return err
+	}
+	normalized := normalizeImages(images)
+	c.Logger().Info(fmt.Sprintf("Found %d images: %s", len(normalized), normalized))
+	return s.streamImages(c, authn, normalized)
+}
+
+func renderHelmChart(reference, version string, values map[string]interface{}, verify bool) (string, error) {
+	// The following code is adapted from https://github.com/helm/helm/blob/master/cmd/helm/install.go
+	client := action.NewInstall(&action.Configuration{})
+	client.ClientOnly = true
+	client.DryRun = true
+	client.Version = version
+	client.Verify = verify
+	name, chart, err := client.NameAndChart([]string{"saveomat", reference})
+	if err != nil {
+		return "", err
+	}
+	client.ReleaseName = name
+
+	cp, err := client.ChartPathOptions.LocateChart(chart, settings)
+	if err != nil {
+		return "", err
+	}
+
+	// Check chart dependencies to make sure all are present in /charts
+	chartRequested, err := loader.Load(cp)
+	if err != nil {
+		return "", err
+	}
+	if req := chartRequested.Metadata.Dependencies; req != nil {
+		// If CheckDependencies returns an error, we have unfulfilled dependencies.
+		// As of Helm 2.4.0, this is treated as a stopping condition:
+		// https://github.com/helm/helm/issues/2209
+		if err := action.CheckDependencies(chartRequested, req); err != nil {
+			if client.DependencyUpdate {
+				man := &downloader.Manager{
+					Out:              os.Stdout,
+					ChartPath:        cp,
+					Keyring:          client.ChartPathOptions.Keyring,
+					SkipUpdate:       false,
+					Getters:          getters,
+					RepositoryConfig: settings.RepositoryConfig,
+					RepositoryCache:  settings.RepositoryCache,
+					Debug:            settings.Debug,
+				}
+				if err := man.Update(); err != nil {
+					return "", err
+				}
+				// Reload the chart with the updated Chart.lock file.
+				if chartRequested, err = loader.Load(cp); err != nil {
+					return "", errors.Wrap(err, "failed reloading chart after repo update")
```

Comment:
```suggestion
					return "", fmt.Errorf("failed reloading chart after repo update: %w", err)
```
Go has now a native way to wrap errors which I prefer. See [fmt.Errorf](https://golang.org/pkg/fmt/#Errorf) and [errors](https://golang.org/pkg/errors/).